enterprise/server/util/oci/oci.go

```diff
@@ -144,39 +151,21 @@ func Resolve(ctx context.Context, imageName string, platform *rgpb.Platform, cre
 		return nil, status.UnavailableErrorf("could not retrieve manifest from remote: %s", err)
 	}
 
-	switch remoteDesc.MediaType {
-	// This is an "image index", a meta-manifest that contains a list of
-	// {platform props, manifest hash} properties to allow client to decide
-	// which manifest they want to use based on platform.
-	case types.OCIImageIndex, types.DockerManifestList:
-		imgIdx, err := remoteDesc.ImageIndex()
-		if err != nil {
-			return nil, status.UnknownErrorf("could not get image index from descriptor: %s", err)
-		}
-		imgs, err := partial.FindImages(imgIdx, match.Platforms(v1.Platform{
-			Architecture: platform.GetArch(),
-			OS:           platform.GetOs(),
-			Variant:      platform.GetVariant(),
-		}))
-		if err != nil {
-			return nil, status.UnavailableErrorf("could not search image index: %s", err)
-		}
-		if len(imgs) == 0 {
-			return nil, status.NotFoundErrorf("could not find suitable image in image index")
-		}
-		if len(imgs) > 1 {
-			return nil, status.NotFoundErrorf("found multiple matching images in image index")
-		}
-		return imgs[0], nil
-	case types.OCIManifestSchema1, types.DockerManifestSchema2:
-		img, err := remoteDesc.Image()
-		if err != nil {
+	img, err := remoteDesc.Image()
+	if err != nil {
+		switch remoteDesc.MediaType {
+		// This is an "image index", a meta-manifest that contains a list of
+		// {platform props, manifest hash} properties to allow client to decide
+		// which manifest they want to use based on platform.
```

Comment:
Can you update this comment to explain that these should be resolved correctly by `Image()` so that's why this is an error?